From 68daa3f01a82834ac23fde53f98e7a2bd7fb1e36 Mon Sep 17 00:00:00 2001
From: Gavin Mak <gavinmak@google.com>
Date: Mon, 29 Jan 2024 21:29:31 +0000
Subject: [PATCH] Use shlex instead of pipes

Importing pipes gets a DeprecationWarning. shlex.quote is equivalent to
pipes.quote. Use that instead.

Bug: 1522874
Change-Id: I18b8598f2e961d58ba71d886921bda9d9ad956ca
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/tools/depot_tools/+/5243707
Reviewed-by: Joanna Wang <jojwang@chromium.org>
Auto-Submit: Gavin Mak <gavinmak@google.com>
Commit-Queue: Joanna Wang <jojwang@chromium.org>
---
Upstream-Status: Backport [https://chromium.googlesource.com/chromium/tools/depot_tools.git/+/b808b1bcdd79b344246e0147e98708b91b4573fa]

 depot_tools/fetch.py         | 6 +++---
 depot_tools/gclient_utils.py | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/depot_tools/fetch.py b/depot_tools/fetch.py
index 969277a286..1fa38c0ec7 100755
--- a/depot_tools/fetch.py
+++ b/depot_tools/fetch.py
@@ -23,7 +23,7 @@ from __future__ import print_function
 import json
 import argparse
 import os
-import pipes
+import shlex
 import re
 import subprocess
 import sys
@@ -62,7 +62,7 @@ class Checkout(object):
     pass
 
   def run(self, cmd, return_stdout=False, **kwargs):
-    print('Running: %s' % (' '.join(pipes.quote(x) for x in cmd)))
+    print('Running: %s' % (' '.join(shlex.quote(x) for x in cmd)))
     if self.options.dry_run:
       return ''
     if return_stdout:
@@ -103,7 +103,7 @@ class GclientCheckout(Checkout):
 class GitCheckout(Checkout):
 
   def run_git(self, *cmd, **kwargs):
-    print('Running: git %s' % (' '.join(pipes.quote(x) for x in cmd)))
+    print('Running: git %s' % (' '.join(shlex.quote(x) for x in cmd)))
     if self.options.dry_run:
       return ''
     return git_common.run(*cmd, **kwargs)
diff --git a/depot_tools/gclient_utils.py b/depot_tools/gclient_utils.py
index 6a1659df34..ee283cc234 100644
--- a/depot_tools/gclient_utils.py
+++ b/depot_tools/gclient_utils.py
@@ -16,7 +16,7 @@ import io
 import logging
 import operator
 import os
-import pipes
+import shlex
 import platform
 import re
 import stat
@@ -384,7 +384,7 @@ def safe_makedirs(tree):
 
 def CommandToStr(args):
   """Converts an arg list into a shell escaped string."""
-  return ' '.join(pipes.quote(arg) for arg in args)
+  return ' '.join(shlex.quote(arg) for arg in args)
 
 
 class Wrapper(object):
